packages/svelte/src/internal/client/render.js

```diff
@@ -36,23 +36,12 @@ export function set_should_intro(value) {
 }
 
 /**
- * @param {Element} dom
+ * @param {Element} text
  * @param {string} value
  * @returns {void}
  */
-export function set_text(dom, value) {
-	// @ts-expect-error need to add __value to patched prototype
-	const prev_node_value = dom.__nodeValue;
-	const next_node_value = stringify(value);
-	if (hydrating && dom.nodeValue === next_node_value) {
-		// In case of hydration don't reset the nodeValue as it's already correct.
-		// @ts-expect-error need to add __nodeValue to patched prototype
-		dom.__nodeValue = next_node_value;
-	} else if (prev_node_value !== next_node_value) {
-		dom.nodeValue = next_node_value;
-		// @ts-expect-error need to add __className to patched prototype
-		dom.__nodeValue = next_node_value;
-	}
+export function set_text(text, value) {
+	text.nodeValue = value;
```

Comment:
Are you removing skipping during hydration because reading `nodeValue` to eventually skip if eventually more expensive than just set the value?

Reply:
there's just nothing to do differently in the hydration case any more

Reply:
> there's just nothing to do differently in the hydration case any more

Wouldn't be possible to skip the assignment since the value is already there from SSR?

Reply:
it could have changed